test/chainrules.jl

```diff
@@ -0,0 +1,45 @@
+@testset "ChainRules" begin
+
+    A = randn(4, 2)
+    D = Diagonal(randn(2).^2 .+ 1)
+    S = Diagonal(randn(4).^2 .+ 1)
+
+    W = WoodburyPDMat(A, D, S)
+    R = 2.0
+    Dmat = Diagonal(rand(4,))
+
+    x = randn(size(A, 1))
+
+    @testset "Constructors" begin
+        test_rrule(WoodburyPDMat, W.A, W.D, W.S)
+        # This is a gradient. Should be able to deal with negative elements
+        test_rrule(WoodburyPDMat, W.A, W.D, W.S;
+            output_tangent=Tangent{WoodburyPDMat}(;
+            A = rand(4,2), D = Diagonal(-1 * rand(2,)), S = Diagonal(-1 * rand(4,)))
+        )
+    end
+
+    @testset "*(Matrix-Woodbury)" begin
```

Comment:
I don't see this rule anywhere, does this work because ChainRules are loaded? might be worth adding a comment

Reply:
Oh I think these are because Zygote is loaded. 

I'll add ChainRules explicitely